contracts/pol/LiquidityManagerCore.sol

```diff
@@ -201,14 +201,34 @@ abstract contract LiquidityManagerCore is ERC721TokenReceiver {
     function _feeAmountTickSpacing(int24 feeTierOrTickSpacing) internal view virtual returns (int24 tickSpacing);
 
     /// @dev Gets sqrt price and observation index values from slot 0.
+    /// @notice This is a general usage function, it accounts for known UniswapV3-like versions of slot0() function.
     /// @param pool Pool address.
     /// @return sqrtPriceX96 Sqrt price.
     /// @return observationIndex Observation index.
     function _getPriceAndObservationIndexFromSlot0(address pool)
         internal
         view
         virtual
-        returns (uint160 sqrtPriceX96, uint16 observationIndex);
+        returns (uint160 sqrtPriceX96, uint16 observationIndex)
+    {
+        // Get slot0 payload
+        bytes memory payload = abi.encodeCall(IUniswapV3.slot0, ());
```

Comment:
`slot0()` has a different set of returns `Uniswap` vs `Aerodrome`, however this function is called for general purpose in checkPoolAndGetCenterPrice(), and cannot only account for either `Uniswap` or `Aerodrome`, must extract values from both.